components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/cache/CacheBackedParDAO.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.par.cache;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.oauth.par.dao.ParMgtDAO;
+import org.wso2.carbon.identity.oauth.par.dao.ParMgtDAOImpl;
+import org.wso2.carbon.identity.oauth.par.exceptions.ParCoreException;
+import org.wso2.carbon.identity.oauth.par.model.ParRequest;
+
+import java.util.HashMap;
+
+/**
+ *
+ * Caching wrapper for org.wso2.carbon.identity.claim.metadata.mgt.dao.ExternalClaimDAO.
+ *
+ */
+public class CacheBackedParDAO implements ParMgtDAO{
+    private static final Log log = LogFactory.getLog(CacheBackedParDAO.class);
+    private final ParCache parCache = ParCache.getInstance();
+    private final ParMgtDAOImpl parMgtDAO = new ParMgtDAOImpl();
+    private final int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
```

Comment:
tenant ID should be a local variable for each request. No point in keeping as global and a final one.